README.md

```diff
@@ -0,0 +1,127 @@
+# Comfygure
+
+Store encrypted project configuration in a centralised place, then check it out and decrypt locally for development, test and deployment.
+
+## The Problem: Managing Application Configuration
+
+How do you store the configuration for a web application? 
+
+Most developers use configuration files (`config.json`, `parameters.yaml`, `root.xml`, etc). But these files should not be committed to source control (git), because they contain sensible information (db passwords, service credentials, etc), and because they can change with each developer. So developers usually commit fake config files (like `parameters.yml-dist`) and keep a real config file locally (and ignored by source control). This leads to problems when one developer adds a new key to the configuration but another developer doesn't know about it.
+
+Another solution is to use environment variables. This makes sharing the configuration between developers and between environments even harder.
+
+## The solution
+
+Comfygure proposes to solve that problem by storing configuration on a remote server (like a remote git), encrypted. A comfygure client knows how to read and write from that remote server, and decrypt the configuration to dump it locally.
+
+![comfygure workflow](./comfy.png)
+
+Ultimately, this lets you execute the following command from any server (development, test, production):
+
+```sh
+> comfy get development --envvars
+LOGIN=admin
+PASSWORD=S3cr3T
+```
+
+Developers store the decryption key locally, allowing them to decrypt and/or update the configuration. In a similar fashion, CI servers can also check out the configuration with a simple decryption key, then build an artifact to be deployed to production.
+
+Comfy handles environments (dev, test, staging, production, etc.), versioning, read/write premissions. It allows to dump the configuration to the format you like (env vars, json, yaml).
+
+From  security standpoint, if the remote server is owned, the attacker will only have the encrypted data. Since the server never stores the key, the attacker will not be able to decrypt it.
```

Comment:
In the database, only the value is encrypted.

For example with:
```json
{
    "app_id": 42,
    "user": { "login": "admin", "password": "S3cr3T" }
}
```
What is stored:

| Key | Value |
|-------|----------|
| `app_id` | encrypted(`42`) |
| `user.login` | encrypted(`admin`) |
| `user.password` | encrypted(`S3cr3T`) |

This will allows us to have an efficient diff (even without the encryption passphrase).
Furthermore, we are able to deduplicate the configurations versions and the choice to display it as a JSON, YAML or env vars.